src/main/java/com/clouway/push/client/ChannelApiPushEventBus.java

```diff
@@ -40,20 +40,15 @@ public HandlerRegistration addHandler(final PushEvent.Type type, final PushEvent
 
     final HandlerRegistration[] handlerRegistration = {null};
 
-    if (!pushChannelApi.hasOpenedChannel()) {
+    if (!pushChannelApi.hasInitialConnection()) {
 
-      pushChannelApi.connect(new AsyncConnectCallback() {
+      pushChannelApi.connect();
```

Comment:
In case when we are unable to open channel api, we have to handle it. As I saw in the API documentation, there are events that need to be handled:

```
onopen
onmessage
onerror
onclose
```

Also the establishConnection method doesn't make anything on failure: 

```java
 pushChannelServiceAsync.connect(subscriber.get(), new AsyncCallback<String>() {

      public void onFailure(Throwable caught) {
      }
}
```